File: lib_mvp/src/main/java/com/kasiengao/mvp/java/BaseContract.java
package com.kasiengao.mvp.java;

import android.content.Context;

import androidx.lifecycle.LifecycleOwner;

/**
 * @ClassName: BaseContract
 * @Author: KaiSenGao
 * @CreateDate: 2020/3/26 16:00
 * @Description:
 */
public interface BaseContract {

    interface BaseView {

        /**
         * 上下文
         *
         * @return Context
         */
        Context getContext();

        /**
         * 显示一个Toast
         *
         * @param message Toast信息
         */
        void showToast(String message);

        /**
         * LifecycleOwner
         *
         * @return LifecycleOwner
         */
        LifecycleOwner getLifecycleOwner();
    }

    interface BasePresenter<V extends BaseView> {

        /**
         * 绑定View
         *
         * @param rootView View
         */
        void attachView(V rootView);

        /**
         * 解绑View
         */
        void detachView();
    }

}

File: lib_mvp/src/main/java/com/kasiengao/mvp/kt/BaseContract.kt
package com.kasiengao.mvp.kt

import android.content.Context
import androidx.lifecycle.LifecycleOwner

/**
 * @ClassName: BaseContract
 * @Author: KaiSenGao
 * @CreateDate: 2020/3/26 17:00
 * @Description:
 */
interface BaseContract {

    interface BaseView {
        /**
         * 上下文
         *
         * @return Context
         */
        val context: Context?

        /**
         * 显示一个Toast
         *
         * @param message Toast信息
         */
        fun showToast(message: String?)

        /**
         * LifecycleOwner
         *
         * @return LifecycleOwner
         */
        val lifecycleOwner: LifecycleOwner?
    }

    interface BasePresenter<V : BaseView> {
        /**
         * 绑定View
         *
         * @param rootView View
         */
        fun attachView(rootView: V)

        /**
         * 解绑View
         */
        fun detachView()
    }
}